pkg/proxy/proxy.go

```diff
@@ -624,6 +660,49 @@ func (s *Proxy) generateToken() string {
 	return fmt.Sprintf("%x", b)
 }
 
+// when external svc becomes active update cluster info
+func (s *Proxy) addSVC(obj interface{}) {
+	svc := obj.(*v1.Service)
+	klog.V(4).Infof("service:%s found", svc.Name)
+	if !strings.HasPrefix(svc.Name, svcPrefix) {
+		klog.V(4).Infof("service:%s has no relation with tower", svc.Name)
+		return
+	}
+	// get cluster by svc name
+	clusterName := strings.TrimLeft(svc.Name, svcPrefix)
+	cluster, err := s.clusterClient.ClusterV1alpha1().Clusters().Get(context.TODO(), clusterName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("get cluster with name[%s] failed. err:%v", clusterName, err)
+	}
+	if len(svc.Status.LoadBalancer.Ingress) > 0 {
+		klog.V(2).Infof("service:%s has ready with external ip", svc.Name)
+		// get cluster by service name
+		var externalAddr string
+		if len(svc.Status.LoadBalancer.Ingress[0].IP) > 0 {
+			externalAddr = svc.Status.LoadBalancer.Ingress[0].IP
+		} else if len(svc.Status.LoadBalancer.Ingress[0].Hostname) > 0 {
+			externalAddr = svc.Status.LoadBalancer.Ingress[0].Hostname
+		}
+		cluster.Spec.Connection.ExternalKubernetesAPIEndpoint = fmt.Sprintf("https://%s:%d", externalAddr, kubernetesPort)
+		cluster, err = s.clusterClient.ClusterV1alpha1().Clusters().Update(context.TODO(), cluster, metav1.UpdateOptions{})
+		if err != nil {
+			klog.Errorf("update cluster:%s error:%v", clusterName, err)
+		} else {
+			externalLBCondition := v1alpha1.ClusterCondition{
+				Type:               v1alpha1.ClusterExternalAccessReady,
+				Status:             corev1.ConditionTrue,
+				Reason:             string(v1alpha1.ClusterInitialized),
+				Message:            "Cluster external kubeapi has been initialized",
+				LastUpdateTime:     metav1.Now(),
+				LastTransitionTime: metav1.Now(),
+			}
+			updateClusterCondition(cluster, externalLBCondition)
+		}
+	} else {
+		klog.V(2).Infof("service:%s has not ready with external ip", svc.Name)
```

Comment:
I think we can skip `else condition`,  because once loadbalancer type service is ready, the user still sees this `not ready` logs, it's confusing. No news is good news.